src/js/mdns_discovery.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const MDNS_INTERVAL = 10000;
+const TCP_CHECK_INTERVAL = 5000;
+const TCP_TIMEOUT = 2000;
+
+const MdnsDiscovery = new function() {
+    this.mdnsBrowser = {
+        services: [],
+        browser: null,
+    };
+
+    this.tcpCheckLock = false;
+};
+
+MdnsDiscovery.initialize = function() {
+    const self = this;
+
+    if (GUI.isCordova()) {
+        const zeroconf = cordova.plugins.zeroconf;
+        zeroconf.registerAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
+        zeroconf.watchAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
+        zeroconf.watch("_http._tcp.", "local.", (result) => {
+            const action = result.action;
+            const service = result.service;
+            console.log("Action",action);
+            console.log("Service",service);
+            if (action==='resolved') {
+                console.log("Zeroconf Service Changed", service);
+                self.mdnsBrowser.services.push({
+                    addresses: service.ipv4Addresses,
+                    txt: service.txtRecord,
+                    fqdn: `${service.name}._http._tcp.local.`,
+                });
+            }
+        });
+    } else {
+        const bonjour = require('bonjour')();
+
+        self.mdnsBrowser.browser = bonjour.find({ type: 'http' }, service => {
+            console.log("Found HTTP service", service);
+            self.mdnsBrowser.services.push({
+                addresses: service.addresses,
+                txt: service.txt,
+                fqdn: service.fqdn,
+            });
+        });
+
+        if (!GUI.isCordova()){
+            self.mdns_timer = setInterval(() => {
+                if (!GUI.connected_to && self.mdnsBrowser.browser) {
+                    self.mdnsBrowser.browser.update();
+                }
+            }, MDNS_INTERVAL);
+        }
+    }
+
+    setTimeout(() => {
```

Comment:
```suggestion
    setInterval(() => {
```

Reply:
Good catch, we didn't copy the timeout which calls itself but this is a nicer sollution